includes/Core/Util/Activation.php

```diff
@@ -155,9 +155,74 @@ private function get_activation_notice() {
 					ob_start();
 					?>
 					<script type="text/javascript">
-					if( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
-						sendAnalyticsTrackingEvent( 'plugin_setup', 'plugin_activated' );
-					}
+						if ( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
+							sendAnalyticsTrackingEvent( 'plugin_setup', 'plugin_activated' );
+						}
+
+						jQuery( document ).ready( function( $ ) {
+							var trackingScriptPresent = !! googlesitekit.admin.trackingOptIn;
+
+							if ( googlesitekit.admin.trackingOptIn ) {
+								$( '#opt-in' ).prop( 'checked', googlesitekit.admin.trackingOptIn );
+							}
+							if ( googlesitekit.admin.proxySetupURL ) {
+								$( '#start-setup-link' ).attr( 'href', googlesitekit.admin.proxySetupURL );
+							}
+
+							$( '#start-setup-link' ).on( 'click' , function() {
+								if ( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
+									sendAnalyticsTrackingEvent( 'plugin_setup', 'goto_sitekit' );
+								}
+							} );
+
+							$('#opt-in').on( 'change' , function( event ) {
+								if ( event.target.disabled ) {
+									event.preventDefault();
+									return;
+								}
+
+								var checked = event.target.checked;
+
+								var body = {
+									googlesitekit_tracking_optin: checked,
+								};
+								var self = this;
+
+								$( self ).prop( 'disabled', true );
+
+								wp.apiFetch( {
+									path: '/wp/v2/settings',
+									headers: {
+										'Content-Type': 'application/json; charset=UTF-8',
+									},
+									body: JSON.stringify( body ),
+									method: 'POST',
+								} )
+									.then( function() {
+										$(self).prop( 'disabled', null );
+
+										var trackingId = googlesitekit.admin.trackingID;
+										var trackingScriptPresent = $( `script[src="https://www.googletagmanager.com/gtag/js?id=${ trackingId }"]` ).length > 0;
```

Comment:
We only need to load the script if they enabled tracking, not if they disabled it. Theoretically we should remove the script if they just disabled it, but I think setting the `window.googlesitekitTrackingEnabled` to `false` will be sufficient since it will then at least no longer fire events (and before then the script will have tracked them already anyway).

Reply:
Yeah, we don't _remove_ the script in other instances here when it's disabled, and I don't think that's needed here. I'll toggle `window.googlesitekitTrackingEnabled` when the setting is posted though, so that should work.